mythril_multiboot2/src/services.rs

```diff
@@ -35,19 +35,29 @@ impl VmServices for Multiboot2Services {
     }
 
     fn rsdp(&self) -> Result<acpi::rsdp::RSDP> {
-        // Get a v1 tag if possible
-        self.info
-            .rsdp_v1_tag()
-            .map(|tag| {
-                // use empty string if no OEM id provided
-                let id = tag.oem_id().unwrap_or("      ").as_bytes();
-                let mut arr: [u8; 6] = [0; 6];
-                arr.copy_from_slice(&id[0..6]);
-                acpi::rsdp::RSDP::V1 {
-                    oemid: arr,
-                    rsdt_addr: tag.rsdt_address() as u32,
-                }
-            })
-            .ok_or(Error::NotFound)
+        let tag_v1 = self.info.rsdp_v1_tag();
+        let tag_v2 = self.info.rsdp_v2_tag();
+
+        if let Some(tag) = tag_v2 {
```

Comment:
You could use [Option::map_or_else](https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or_else) with the RSDPv1 as the default and the RSDPv2 as the main function or like the current sollution a `map` after attempting to find the `rsdp_v2_tag`, followed by an [Option::ok_or_else](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else) for the `rsdp_v1_tag`, followed by an `ok_or(Error::NotFound)`

Reply:
I attempted to do so. Used Option::map_or_else a couple times

Reply:
let me know what you think